main.go

```diff
@@ -101,6 +101,17 @@ func newHTTPProxy(cfg *config.Config) http.Handler {
 				KeepAlive: cfg.Proxy.KeepAliveTimeout,
 			}).Dial,
 		},
+		InsecureTransport: &http.Transport{
```

Comment:
Maybe move this into a local `newTransport := func(tlscfg *tls.Config) http.Transport` func to avoid the duplication.